### ISSUE TITLE
StreamingJob broken with azure native provider

### ISSUE THREAD

**Issue by WillooWisp** [pulumi/pulumi-azure-native#688]
Using the separate Input and Output resource components ​together ​with the StreamingJob resource, makes up fail with the error below. It works with azure-nextgen v0.5.1 but not with latest azure-native provider and default namespace.

error: Code="422" Message="The streaming job should have at least one input defined." Details=[{"code":"422","correlationId":"","message":"The streaming job should have at least one input defined.","requestId":""}]
@ updating...|

```
var streamingJob = new StreamingJob("sasj-123456", jobArgs, new CustomResourceOptions());

new Input($"sai-{jobName}-{inputName}",
    ​new InputArgs
    ​{
        ​Properties = new StreamInputPropertiesArgs
        ​{
            ​Datasource = datasource,
            ​Serialization = serialization,
            ​Type = "Stream"
        ​},
        ​Name = inputName,
        ​JobName = streamingJob.Name,
        ​InputName = inputName,
        ​ResourceGroupName = resourceGroupName,
    ​},
    ​new CustomResourceOptions()
​);
```

**Comment by mikhailshilkov** [comment]
I checked that 0.5.1 and 0.7.1 use the same API version 2016-03-01, so the difference in behavior is surprising.

Do you have a full repro that I could run? Do you still have the working version?

**Comment by WillooWisp** [comment]
Yes, after a lot if testing to see what I more had changed to get the faulty behavior I finally found this change causing the error.

Setting the property OutputStartMode on the StreamingJob arguments to something, e.g. OutputStartMode.LastOutputEventTime, causes this very confusing error.

**Comment by WillooWisp** [comment]
I found an even worse scenario, where all inputs and outputs are removed upon pulumi up, without any info about that any resources are going to be deleted (happens after streaming job has been started at least once and then stopped).

1. Make an up with a streaming job and an input and output.
2. Start the streaming job, e.g. in Azure portal.
3. Wait for it to start.
4. Stop the streaming job, needed in order to not fail again on next up.
5. Perform a pulumi up command, shows this info in the diff...
```
~ azure-native:streamanalytics:StreamingJob: (update)
      - outputStartMode: "JobStartTime"
      - outputStartTime: "2021-04-06T13:59:18.357Z"
```
6. All input and output resources are now missing from the streaming job.
7. Make a pulumi refresh and the state will be updated by deleting the resources no longer in Azure.
8. Process can be repeated.

**Comment by mikhailshilkov** [comment]
Did you run refresh or change code before (5)? If not, why does it show any diff?

**Comment by WillooWisp** [comment]
You are correct of course, added the missing refresh step before up, but still, it should not remove any resources.

**Comment by mikhailshilkov** [comment]
This sounds like the same class of issue as https://github.com/pulumi/pulumi-azure-native/issues/611 caused by unfortunately parent-child Azure resource design

**Comment by WillooWisp** [comment]
So no recommended work around or alternate configuration setup to make it work?

**Comment by mikhailshilkov** [comment]
Same as for vnet-subnet, two options:

1. Define inputs and outputs inline inside the `StreamingJob` resource.
2. Add them to `IgnoreChanges` and never update the `StreamingJob`.

FWIW, I suspect you'd have the same issue with ARM templates.

**Comment by WillooWisp** [comment]
Okay, one question though...

The initial problem reported at the top here was that I got a weird behavior where pulumi up fails, deploying resources from scratch, with error about no inputs being defined for the streaming job, when the property OutputStartMode is set. Feels kind of related to me, since depending on the state of the output start mode (which is also changed / synced on refresh) it ignores all inputs and outputs? Just want to make sure there is nothing strange going in that can be sorted out differently. What do you think @mikhailshilkov?

**Comment by mikhailshilkov** [comment]
All I can say is that that error comes from the Azure API. I've never used these resources personally so I don't know the semantics here. If that's helpful, you can always run `pulumi up` with debug flag and see all the HTTP requests and responses in the logs to see what goes over the wire.

**Comment by WillooWisp** [comment]
Refactoring the code to use inline inputs and outputs as part of streaming job solves the issue, even though it does not seem to refresh changes made to inputs and outputs in Azure and therefore no diff is shown, but I guess you have to live with one of the problems.
